ion/timestamp_test.go

```diff
@@ -208,3 +225,148 @@ func TestTimestampString(t *testing.T) {
+			expected: 1,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 220000000, TimestampPrecisionNanosecond, 1},
+			name:     "2000-01-02T03:04:05.2Z",
+			expected: 2,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 12000000, TimestampPrecisionNanosecond, 1},
+			name:     "2000-01-02T03:04:05.0Z",
+			expected: 0,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 12000000, TimestampPrecisionNanosecond, 2},
+			name:     "2000-01-02T03:04:05.01Z",
+			expected: 1,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 120000000, TimestampPrecisionNanosecond, 2},
+			name:     "2000-01-02T03:04:05.12Z",
+			expected: 12,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123000000, TimestampPrecisionNanosecond, 3},
+			name:     "2000-01-02T03:04:05.123Z",
+			expected: 123,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123456789, TimestampPrecisionNanosecond, 4},
+			name:     "2000-01-02T03:04:05.1234Z",
+			expected: 1234,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123450000, TimestampPrecisionNanosecond, 5},
+			name:     "2000-01-02T03:04:05.12345Z",
+			expected: 12345,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123456000, TimestampPrecisionNanosecond, 6},
+			name:     "2000-01-02T03:04:05.123456Z",
+			expected: 123456,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123456700, TimestampPrecisionNanosecond, 7},
+			name:     "2000-01-02T03:04:05.1234567Z",
+			expected: 1234567,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123456780, TimestampPrecisionNanosecond, 8},
+			name:     "2000-01-02T03:04:05.12345678Z",
+			expected: 12345678,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 123456789, TimestampPrecisionNanosecond, 9},
+			name:     "2000-01-02T03:04:05.123456789Z",
+			expected: 123456789,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 5000, TimestampPrecisionNanosecond, 2},
+			name:     "2000-01-02T03:04:05.000005000",
+			expected: 0,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 6000, TimestampPrecisionNanosecond, 5},
+			name:     "2000-01-02T03:04:05.000006000",
+			expected: 0,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 7000, TimestampPrecisionNanosecond, 6},
+			name:     "2000-01-02T03:04:05.000007000",
+			expected: 7,
+		},
+		{
+			fields:   fields{2000, 1, 2, 3, 4, 5, 7001, TimestampPrecisionNanosecond, 6},
+			name:     "2000-01-02T03:04:05.000007001",
+			expected: 7,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dateTime := time.Date(tt.fields.year, time.Month(tt.fields.month), tt.fields.day, tt.fields.hour,
+				tt.fields.minute, tt.fields.second, tt.fields.nanosecond, time.UTC)
+
+			kind := TimezoneUnspecified
+			if tt.fields.precision >= TimestampPrecisionMinute {
+				kind = TimezoneUTC
+			}
+
+			ts := &Timestamp{
+				dateTime:             dateTime,
+				precision:            tt.fields.precision,
+				kind:                 kind,
+				numFractionalSeconds: tt.fields.numFractionalSeconds,
```

Comment:
Am I missing something, or should this field have a name like `fractionalSecondsDecimalPlaces`? (i.e. It's a number of decimal places, not a number of fractional seconds.)

Reply:
That is correct.
* `timestamp, err := ParseTimestamp("2006-01-02T15:04:05.123Z")` will have timestamp.numFractionalSeconds = 3
* `timestamp, err := ParseTimestamp("2006-01-02T15:04:05.123000Z")` will have timestamp.numFractionalSeconds = 6

Reply:
Thanks for clarifying. I find this naming confusing, but it's part of the public API so we can't change it without a major version bump.